### DISCUSSION TITLE
Rectangle Pro Settings dialog opens on desktop after Mac start up

### DISCUSSION THREAD

**Discussion by REmenecker-SR** [rxhanson/RectanglePro-Community#484] [Q&A] [answered]
When booting my Mac, Rectangle Pro's _Settings_ window opens after the app starts at login (see attach screenshot). I cannot recall specifically if this started happening after upgrading to version 3 or one of the version 3 maintenance updates (3.0.x). I want to have Rectangle Pro launch when I login, but don't want to have to keep closing the Settings window. 

Environment:
* MacBook Pro (16-inch, 2019 Intel)
* macOS Ventura 13.6.5
* Rectangle Pro 3.0.21

I have not yet reinstalled or tried resetting the app. I first wanted to check here to see if anyone else has encountered this and how they resolved it. 

![Screenshot 2024-04-03 at 8 13 06 AM](https://github.com/rxhanson/RectanglePro-Community/assets/83981062/8137098f-6282-42b9-9c9a-44ff6aff98b5)

**Comment by rxhanson** [comment, marked as answer]
The Settings window appears when the app is launched and it's already running. It's likely that you have some sort of scenario where launch on login is occurring twice. Maybe try disabling launch on login in the app, then checking to see if it's in the Login Items in System Settings?

**Reply by REmenecker-SR** [reply]
Thanks @rxhanson!

After reading your reply, I dug a bit further and found two _Login Items_ entries: One in _Open at Login_ and a second under _Allow in Background._ The entry in _Open at Login_ would follow the _Launch on login_ setting in the app's settings dialog. Disabling the _Allow in Background_ entry stopped the duplicate launching of the app at login.

**Reply by rxhanson** [reply]
Great! Apple released an API for "Open at Login" in Ventura, which was adopted for Rectangle Pro at that time. There was logic in Rectangle Pro to remove the "Allow in Background" toggle, but I think there might have been a macOS bug that didn't consistently remove it there programmatically 100% of the time. Perhaps you had hit some form of that macOS bug at some point.

**Reply by REmenecker-SR** [reply]
> Perhaps you had hit some form of that macOS bug at some point.

That would make sense. It would not be the first time. I have a short list of odd bugs with 3rd-party utilities that I cannot replicate in a test user account I set up on my Mac. 

Thanks for helping me get my bearings on resolving the issue!
